### PR TITLE
Add byte_* to Docs

### PR DESCRIPTION
**Pull request by x62smith**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues/2081) ticket:

Describe changes:
- added byte_test
- added byte_jump
- added byte_extract

I used tables but if you'd like it done with pictures instead please let me know.

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: doc/userguide/rules/payload-keywords.rst [hunk 1]
```diff
@@ -264,6 +264,169 @@ example of dsize in a rule:
 
     alert udp $EXTERNAL_NET any -> $HOME_NET 65535 (msg:"GPL DELETED EXPLOIT LANDesk Management Suite Alerting Service buffer overflow"; :example-rule-emphasis:`dsize:>268;` reference: bugtraq,23483; reference: cve,2007-1674; classtype: attempted-admin; sid:100000928; rev:1;)
 
+byte_test
+---------
+The ``byte_test`` keyword extracts ``<num of bytes>`` and performs an opertion selected with ``<operator>`` against the value in ``<test value>`` at a particular ``<offset>``.
```
Comment (norg) [inline doc/userguide/rules/payload-keywords.rst]: oper**a**tion

File: doc/userguide/rules/payload-keywords.rst [hunk 2]
```diff
@@ -264,6 +264,169 @@ example of dsize in a rule:
...
+|		 | - [!] Negation can prefix other operators					|
+|	         | - < less than								|
+|      	         | - > greater than								|
+|		 | - = equal									|
+|		 | - <= less than or equal							|
+|		 | - >= greater than or equal							|
+|		 | - & bitwise AND								|
+|		 | - ^ bitwise OR								|
++----------------+------------------------------------------------------------------------------+
+| <value>	 | Value to test teh converted value against [hex or decimal accepted]		|
```
Comment (norg) [inline doc/userguide/rules/payload-keywords.rst]: **the** instead of **teh**

File: doc/userguide/rules/payload-keywords.rst [hunk 3]
```diff
@@ -264,6 +264,169 @@ example of dsize in a rule:
...
+	 content:"|00 01 00 02|"; byte_test:2,&,0x01,relative;)
+
+  alert tcp any any -> any any (msg:"Byte_Test Example - Compare to String"; \
+ 	 content:"foobar"; byte_test:4,=,1337,1,relative,string,dec;)
+
+
+byte_jump
+---------
+
+The ``byte_jump`` keyword alows for the ability to select a ``<num of bytes>`` from an ``<offset>`` and moves the detection pointer to that position.  Content matches will then be based off the new position.
```
Comment (norg) [inline doc/userguide/rules/payload-keywords.rst]: al**l**ows

File: doc/userguide/rules/payload-keywords.rst [hunk 4]
```diff
@@ -264,6 +264,169 @@ example of dsize in a rule:
...
++-----------------------+-----------------------------------------------------------------------+
+| [post_offset] <value>	| After the jump operation has been performed, it will			|
+|			| jump an additional number of bytes specified by <value>		|
++-----------------------+-----------------------------------------------------------------------+
+| [dce]			| Allow the DCE module determine the byte order.			|
++-----------------------+-----------------------------------------------------------------------+
+| [bitmask] <value>	| The AND operator will be applied by <value> and the			|
+|			| converted bytes, then jump operation is performed.			|
++-----------------------+-----------------------------------------------------------------------+
+
```
Comment (norg) [inline doc/userguide/rules/payload-keywords.rst]: you end some with a **.** and some not :) maybe make it consistent

**Comment by victorjulien** [comment]
Other than @norg's comments I would like to ask you to target this against the 'master' branch instead of the master-4.1.x branch.

Please make the requested changes in a new branch and then open a new PR. See the workflow here https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Github_work_flow#Create-a-new-branch-for-incorporating-feedback

**Comment by victorjulien** [comment]
Replaced by #3701

**Pull request closed without merging**